File: backend/cs/SocialNetApp/Dialogs/Core/Model/Message.cs
using Microsoft.AspNetCore.Mvc.RazorPages;
using System.Text.Json.Serialization;

namespace Dialogs.Core.Model
{
    public readonly record  struct Message
    {
        [JsonPropertyName("id")]
        public uint Id { get; init; }

        [JsonPropertyName("authorId")]
        public uint AuthorId { get; init; }

        [JsonPropertyName("userId")]
        public ulong UserId { get; init; }

        [JsonPropertyName("message")]
        public string Text { get; init; }

        [JsonPropertyName("created")]
        public DateTime? Created { get; init; }
    }
}

File: backend/go/socialnetapp/websockets/internal/model/message.go
package model

import "time"

// Message represents model for a dialog message
type Message struct {
	Id       uint       `json:"id"`
	AuthorId uint       `json:"authorId"`
	UserId   uint       `json:"userId"`
	Text     string     `json:"message"`
	Created  *time.Time `json:"created"`
}
